pkg/storage/stores/shipper/compactor/deletion/tenant_delete_requests_client.go

```diff
@@ -0,0 +1,33 @@
+package deletion
+
+import (
+	"context"
+
+	"github.com/grafana/loki/pkg/storage/stores/shipper/compactor/retention"
+)
+
+type perTenantDeleteRequestsClient struct {
+	client DeleteRequestsClient
+	limits retention.Limits
+}
+
+func NewPerTenantDeleteRequestsClient(c DeleteRequestsClient, l retention.Limits) DeleteRequestsClient {
+	return &perTenantDeleteRequestsClient{
+		client: c,
+		limits: l,
+	}
+}
+
+func (c *perTenantDeleteRequestsClient) GetAllDeleteRequestsForUser(ctx context.Context, userID string) ([]DeleteRequest, error) {
+	allLimits := c.limits.AllByUserID()
+	userLimits, ok := allLimits[userID]
+	if ok && userLimits.CompactorDeletionEnabled || c.limits.DefaultLimits().CompactorDeletionEnabled {
+		return c.client.GetAllDeleteRequestsForUser(ctx, userID)
+	}
+
+	return nil, nil
+}
+
+func (c *perTenantDeleteRequestsClient) Stop() {
```

Comment:
Here "Stop" is implemented but it is only used in the tests so I suppose you missed to use it somewhere in the code.

Reply:
Stop is used in the listener that's created in `modules.go`: https://github.com/grafana/loki/blob/f6ba37099bf8791e819881821e6c4309a27bd6cb/pkg/loki/modules.go#L322

https://github.com/grafana/loki/blob/f6ba37099bf8791e819881821e6c4309a27bd6cb/pkg/loki/delete_store_listener.go#L29

It's implemented here so this satisfies the `DeleteRequestClient` interface and just passes through to real client